File: service/db/src/main/java/org/delivery/db/order/UserOrderRepository.java
package org.delivery.db.order;

import java.util.List;
import java.util.Optional;
import org.delivery.db.order.vo.UserOrderStatus;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserOrderRepository extends JpaRepository<UserOrder, Long> {

    List<UserOrder> findAllByUserIdAndStatusOrderByIdDesc(final Long userId, final UserOrderStatus status);

    List<UserOrder> findAllByUserIdAndStatusInOrderByIdDesc(final Long userId, final List<UserOrderStatus> status);

    Optional<UserOrder> findAllByIdAndStatusAndUserId(final Long id, final UserOrderStatus status, final Long userId);

}

File: delivery/service/db/src/main/kotlin/org/delivery/db/order/UserOrderRepository.kt
package org.delivery.db.order

import org.delivery.db.order.vo.UserOrderStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserOrderRepository : JpaRepository<UserOrder, Long> {

    fun findAllByUserIdAndStatusOrderByIdDesc(userId: Long?, status: UserOrderStatus?): List<UserOrder>

    fun findAllByUserIdAndStatusInOrderByIdDesc(userId: Long?, status: List<UserOrderStatus>): List<UserOrder>

    fun findAllByIdAndStatusAndUserId(id: Long?, status: UserOrderStatus?, userId: Long?): UserOrder?

    fun findAllByIdAndUserId(id: Long?, userId: Long?): UserOrder?

}